src/main/java/com/nirmata/workflow/details/WorkflowListenerManagerImpl.java

```diff
@@ -21,55 +21,48 @@
 import com.nirmata.workflow.models.RunId;
 import com.nirmata.workflow.models.TaskId;
 import org.apache.curator.framework.listen.Listenable;
-import org.apache.curator.framework.listen.ListenerContainer;
+import org.apache.curator.framework.listen.StandardListenerManager;
+import org.apache.curator.framework.listen.UnaryListenerManager;
 import org.apache.curator.framework.recipes.cache.PathChildrenCache;
 import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
 import org.apache.curator.utils.CloseableUtils;
+
 import java.io.IOException;
 
-public class WorkflowListenerManagerImpl implements WorkflowListenerManager
-{
+public class WorkflowListenerManagerImpl implements WorkflowListenerManager {
     private final PathChildrenCache completedTasksCache;
     private final PathChildrenCache startedTasksCache;
     private final PathChildrenCache runsCache;
-    private final ListenerContainer<WorkflowListener> listenerContainer = new ListenerContainer<>();
+    private final UnaryListenerManager<WorkflowListener> listenerContainer = StandardListenerManager.standard();
```

Comment:
Note: changes here due to API changes in Curator. This is not directly related to the issue being fixed in this PR.